File: app/src/main/java/de/christinecoenen/code/zapp/model/IChannelList.java
package de.christinecoenen.code.zapp.model;

import java.util.List;

public interface IChannelList extends Iterable<ChannelModel> {

	ChannelModel get(int index);
	ChannelModel get(String id);
	int size();
	List<ChannelModel> getList();
	int indexOf(String channelId);
}

File: app/src/main/java/de/christinecoenen/code/zapp/model/IChannelList.kt
package de.christinecoenen.code.zapp.model

interface IChannelList : Iterable<ChannelModel> {

	val list: List<ChannelModel>

	operator fun get(index: Int): ChannelModel
	operator fun get(id: String): ChannelModel
	fun size(): Int
	fun indexOf(channelId: String): Int

}
